src/Services/StoreClient.php

```diff
@@ -219,11 +222,18 @@ public function createInAppPurchaseCart(string $extensionName, string $feature,
             throw ExtensionStoreException::createStoreApiExceptionFromClientError($e);
         }
 
-        return InAppPurchaseCartStruct::fromArray(json_decode((string) $response->getBody(), true));
+        $inAppPurchaseCart = InAppPurchaseCartStruct::fromArray(json_decode((string) $response->getBody(), true));
+        $inAppPurchaseCart->getPositions()->map(function (InAppPurchaseCartPositionStruct $position) use ($extensionName): void {
+            if ($position->getExtensionName() === '') {
+                $position->setExtensionName($extensionName);
+            }
```

Comment:
Really pleased if we could do
```suggestion
            $position->setExtensionName($position->getExtensionName() ?: $extensionName);
```